### PR TITLE
Add feature toggle for search box on work pages

### PR DESCRIPTION
**Pull request by davidpmccormick**
Next thing ready for A/B testing is having the search box at the top of single work pages. This PR gets that behind a feature toggle so we can check it internally before promoting it to an A/B test.

screenshot 2019-01-17 at 15 01 47

### CODE REVIEW

File: catalogue/webapp/components/SearchForm/SearchForm.js [hunk 1]
```diff
@@ -113,9 +113,9 @@ const SearchForm = ({
                   action: 'clear search',
                   label: 'works-search'
                 });
-                const link = worksUrl({query: null, page: null});
+
+                // TODO: re-focus input (useRef?)
                 setQuery('');
-                Router.push(link.href, link.as);
```
Comment (jamesgorrie) [inline catalogue/webapp/components/SearchForm/SearchForm.js]: How is the route changed?

File: catalogue/webapp/pages/work.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // @flow
 import type {Work, CatalogueApiError, CatalogueApiRedirect} from '../services/catalogue/works';
-import {Fragment} from 'react';
+import {Fragment, useState} from 'react';
```
Comment (jamesgorrie) [inline catalogue/webapp/pages/work.js]: Is this used?

Comment (davidpmccormick) [reply catalogue/webapp/pages/work.js]: don't think so – thanks

File: catalogue/webapp/components/SearchForm/SearchForm.js (hunk 1 above)
Comment (davidpmccormick) [reply catalogue/webapp/components/SearchForm/SearchForm.js]: It isn't – when we clear out the search box by clicking this button, we don't want to change the route, we just want an empty input. Currently, you go back to the /works landing page, which feels broken.

**Comment by Heesoomoon** [comment]
'Search result' link feels like it should be part of Search component. Could we move it below the search bar inside the beige box?

**Comment by jamesgorrie** [comment]
> 'Search result' link feels like it should be part of Search component. Could we move it below the search bar inside the beige box?

Could we run the A/B test so as not to complicate the implementation? I prefer to have as few moving parts on A/B tests so we don't introduce any bugs... we'll definitely get it done if the test proves positive.

File: catalogue/webapp/components/SearchForm/SearchForm.js (hunk 1 above)
Comment (jamesgorrie) [reply catalogue/webapp/components/SearchForm/SearchForm.js]: Ah, sorry, my bad, didn't realise it was on the `ClearSearch`.

File: catalogue/webapp/components/SearchForm/SearchForm.js [hunk 3]
```diff
@@ -113,9 +113,9 @@ const SearchForm = ({
                   action: 'clear search',
                   label: 'works-search'
                 });
-                const link = worksUrl({query: null, page: null});
+
+                // TODO: re-focus input (useRef?)
```
Comment (jamesgorrie) [inline catalogue/webapp/components/SearchForm/SearchForm.js]: When do we think we should refocus?

**Comment by jamesgorrie** [comment]
For reference and to keep the metric linked #3956

**Comment by davidpmccormick** [comment]
> Could we run the A/B test first so as not to complicate the implementation? I prefer to have as few moving parts on A/B tests so we don't introduce any bugs... we'll definitely get it done if the test proves positive.

Ah – I was inclined to agree initially, but didn't feel like too much heavy lifting to change this style _just_ for the B condition.

What do we think (or should I revert last commit)?

screenshot 2019-01-17 at 16 56 15

**Comment by jamesgorrie** [comment]
> What do we think (or should I revert last commit)?

Not too fussed, just a philosophical point of only change what we're testing.

**Pull request merged**